unstructured_ingest/v2/processes/connectors/redisdb.py

```diff
@@ -0,0 +1,209 @@
+                f"at {self.connection_config.host}",
+            )
+            client = self.create_client(async_flag=False)
+            pipeline = client.pipeline()
+            first_element = elements_dict[0]
+            element_id = first_element["element_id"]
+            redis_stack = True
+            try:
+                pipeline.json().set(element_id, "$", first_element)
+                pipeline.execute()
+            except redis_exceptions.ResponseError:
+                # if redis server doesn't have stack extension,
+                # then cannot save as JSON type, save as string instead
+                pipeline.set(element_id, json.dumps(first_element))
+                pipeline.execute()
+                redis_stack = False
+
+            for chunk in batch_generator(elements_dict[1:], self.upload_config.batch_size):
+                for element in chunk:
+                    element_id = element["element_id"]
+                    if redis_stack:
+                        pipeline.json().set(element_id, "$", element)
+                    else:
+                        pipeline.set(element_id, json.dumps(element))
+                pipeline.execute()
+            client.close()
+
+    @requires_dependencies(["redis"], extras="redis")
+    async def run_async(self, path: Path, file_data: FileData, **kwargs: Any) -> None:
+        from redis import exceptions as redis_exceptions
+
+        with path.open("r") as file:
+            elements_dict = json.load(file)
+
+        if elements_dict:
+            logger.info(
+                f"writing {len(elements_dict)} objects to destination asynchronously, "
+                f"db, {self.connection_config.database}, "
+                f"at {self.connection_config.host}",
+            )
+            client = await self.create_client(async_flag=True)
+            async with client.pipeline(transaction=True) as pipe:
+                first_element = elements_dict[0]
+                element_id = first_element["element_id"]
+                redis_stack = True
+                try:
+                    await pipe.json().set(element_id, "$", first_element).execute()
+                except redis_exceptions.ResponseError:
+                    # if redis server doesn't have stack extension,
+                    # then cannot save as JSON type, save as string instead
+                    await pipe.set(element_id, json.dumps(first_element)).execute()
+                    redis_stack = False
+
+                for chunk in batch_generator(elements_dict[1:], self.upload_config.batch_size):
+                    for element in chunk:
+                        element_id = element["element_id"]
+                        if redis_stack:
+                            pipe.json().set(element_id, "$", element)
+                        else:
+                            pipe.set(element_id, json.dumps(element))
+                    await pipe.execute()
+            await client.aclose()
```

Comment:
Given that Redis client has a `close` implement the client providing method as a context manager, check [how qdrant does it](https://github.com/Unstructured-IO/unstructured-ingest/blob/0dc113ca963fe543297bb22a50c0e2f0e029c2d2/unstructured_ingest/v2/processes/connectors/qdrant/qdrant.py#L45) for reference